### DISCUSSION TITLE
something wrong with --labels

### DISCUSSION THREAD

**Discussion by Rachelyolo** [CoBrALab/RABIES#379] [Q&A]
Hi,

I'm new to RABIES and I'm trying to preprocess rat image.

This is the command used. :
![Screenshot from 2024-08-21 17-34-38](https://github.com/user-attachments/assets/2e60afdc-c2aa-40c2-822f-b098988f7dcf)
 
Actually, the file path is correct, I don't know why It always showed  'No such file or directory'.  Could you help me with this, thank you!

**Comment by Gab-D-G** [comment]
Hi, a file path does not end with '/', try removing this. Also the '=' is can be replaced by a space, I believe both can work. Otherwise double check the path is accurate.

Best,
Gabriel

**Reply by Rachelyolo** [reply]
Thank you Gabriel ! 

I solved this problem according to your advice. But I found that the "CSF mask" still has problems. Based on the previous post[#370](https://github.com/CoBrALab/RABIES/discussions/370), I realized that the voxel size of the original SIGMA rat CSF mask was not correct, so I resliced this mask to make it 0.15x0.15x0.15, but the error shows "it is not a binary mask".
![Screenshot from 2024-08-22 11-10-27](https://github.com/user-attachments/assets/9b13fcd3-8f89-41f1-9df0-364238326f4d)
I checked the information of this mask using matlab, and it only has values of 0 and 1, and there are no NaN value either. I don't know why it still prompts "it is not a binary mask."
Could you help me with this? Thank you very much！

Best,
Qiuyi

**Reply by Gab-D-G** [reply]
Hello Qiuyi,
The function raising this is error is looking for any value other than 0 or 1 in the file https://github.com/CoBrALab/RABIES/blob/16cc9da0c91c7ca17cdfd72f1ca3c5e20d0be888/rabies/run_main.py#L340. I am wondering why there is this discrepancy between matlab and our python script. Perhaps the values are stored as floats instead of integers? This may need further inspections.

Best,
Gabriel

**Reply by gdevenyi** [reply]
> I solved this problem according to your advice. But I found that the "CSF mask" still has problems.

Is this a mask we supply, or did you get it from somewhere else.

Please open an issue fully describing your issue with the mask specifically separate from this discussion.

**Reply by Rachelyolo** [reply]
Thank you for your reply. After checking with the CSF mask, I have solved this problem.  However, when I tried to preprocess multiple datasets, the process encountered an error after successfully processing 3 of the datasets. The error message in the log file showed that "Workflow did not execute cleanly. Check log for details". 
![Screenshot from 2024-08-26 20-25-34](https://github.com/user-attachments/assets/e7705f21-6969-4da2-ad8b-c87820973894)
Could you help me with this? Thank you very much!

**Reply by Gab-D-G** [reply]
Hi, the error will be found in the log file, you should be able to the specific problem if you search for 'error'. Please share the log file for us to help.

Gabriel

**Reply by Rachelyolo** [reply]
Here is the log file. [rabies_preprocess.log](https://github.com/user-attachments/files/16755568/rabies_preprocess.log)
Thank you very much!

Best,
Qiuyi

**Reply by Gab-D-G** [reply]
You are getting <Signals.SIGBUS: 7> errors at various pipeline stages, e.g.:
```
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'antsMotionCorrStats -m /preprocess/main_wf/bold_main_wf/bold_hmc_wf/_scan_info_subject_idLPS.session3_split_name_sub-LPS_ses-3_T2w/_run_2/ants_MC/ants_mc_tmp/motcorrMOCOparams.csv -o motcorr_vol19.mat -t 19' died with <Signals.SIGBUS: 7>.
```
Other users have encountered this issue #349 , but this is not a RABIES error, rather it has something to do with your computing platform. You may want to review resource allocation when running the pipeline. You could attempt running the pipeline with -p Linear and see if that mitigates the issue (although it will be slower to run).
